src/main/java/InfrastructureManager/Modules/NetworkStructure/Network.java

```diff
@@ -237,50 +236,222 @@ else if(update == Network.Update.DELETE) {
 	}
 
 	/**
-	 * Getter function
-	 * @return applicationDeviceList
+	 * Getter function to get a particular ApplicationInstanceDeviceRelation from applicationDeviceList
+	 * @param id
+	 * @return ApplicationInstanceDeviceRelation
 	 */
-	public List<ApplicationInstanceDeviceRelation> getApplicationDeviceList() {
-		return applicationDeviceList;
+	public ApplicationInstanceDeviceRelation getApplicationInstanceDeviceRelationFromList(String id)	{
```

Comment:
I don't understand what exactly this function does. I mean what happens if we have an application and a device with the same id?
There are other similar places in this class.